### PR TITLE
Missing fixes from DBIntegrity

### PR DESCRIPTION
**Pull request by morganbr**
I found a set of fixes that I somehow didn't push with DBIntegrity. This should fix team creation, editing and deletion, puzzle deletion, and feedback submission.

### CODE REVIEW

**Comment by morganbr** [comment]
@tabascq, I know you had some of the team fixes in #248, but I realized there were more bugs when I found my change.

File: ServerCore/Pages/Teams/Delete.cshtml.cs [hunk 1]
```diff
@@ -37,7 +37,7 @@ public async Task<IActionResult> OnPostAsync(int teamId)
                 await TeamHelper.DeleteTeamAsync(_context, Team);
             }
 
-            return RedirectToPage("./Index");
+            return RedirectToPage("./List");
```
Comment (tabascq) [inline ServerCore/Pages/Teams/Delete.cshtml.cs]: is this destination right for admins as well? Other places have needed to check which EventRole is active and I'm not sure if this is one of them.

Comment (morganbr) [reply ServerCore/Pages/Teams/Delete.cshtml.cs]: Changed to send admins to index (list works too, but index is more useful)

**Comment by morganbr** [comment]
@tabascq, yep you can just close that one out.

**Comment by morganbr** [comment]
Thanks for the review!

**Pull request merged**